### PR TITLE
feat(lambda): add getTraceUrl function to EpsagonHandler

### PR DESCRIPTION
**Pull request by idonava**

### CODE REVIEW

File: src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs [hunk 1]
```diff
@@ -28,20 +31,20 @@ public class EpsagonHandler
             {
 
                 Utils.DebugLogIfEnabled("entered epsagon lambda handler");
-
+
```
Comment (tvaintrob) [inline src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs:34]: trim whitespaces, all files

File: src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs [hunk 2]
```diff
@@ -244,5 +249,11 @@ private static void HandleInstrumentationError(Exception ex)
 
             InstumentationExceptionsCollector.Exceptions.Add(ex);
         }
+        private static void setTraceUrl(string partialTraceUrl)
```
Comment (tvaintrob) [inline src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs:252]: functions are PascalCase in c# please update all relevant funcs

File: src/Epsagon.Dotnet.Lambda/LambdaTriggerHandler.cs [hunk 3]
```diff
@@ -27,21 +28,26 @@ public void HandleBefore()
...
-            this.scope.Span.SetTag("aws.lambda.aws_account", this.context.InvokedFunctionArn.Split(':')[AWS_ACCOUNT_INDEX]);
-            this.scope.Span.SetTag("aws.lambda.region", Environment.GetEnvironmentVariable("AWS_REGION"));
+            this.scope.Span.SetTag("aws.lambda.aws_account", awsAccount);
+            this.scope.Span.SetTag("aws.lambda.region", awsRegion);
             this.scope.Span.SetTag("aws.lambda.memory", this.context.MemoryLimitInMB.ToString());
             this.scope.Span.SetTag("aws.lambda.function_version", this.context.FunctionVersion);
             this.scope.Span.SetTag("aws.lambda.log_group_name", this.context.LogGroupName);
             this.scope.Span.SetTag("aws.lambda.log_stream_name", this.context.LogStreamName);
             this.scope.Span.SetTag("aws.lambda.cold_start", coldStart);
+            traceUrl = $"https://app.epsagon.com/functions/{awsAccount}/{awsRegion}/{this.context.FunctionName}?requestId={awsRequestId}";
```
Comment (tvaintrob) [inline src/Epsagon.Dotnet.Lambda/LambdaTriggerHandler.cs:50]: access local vars using `this` for consistency and clarity

File: src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs [hunk 4]
```diff
@@ -28,20 +31,20 @@ public class EpsagonHandler
...
                 }
-
+       
                 // handle invocation event
                 using (var scope = GlobalTracer.Instance.BuildSpan((typeof(TEvent).Name)).StartActive(finishSpanOnDispose: true))
                 using (var handler = new LambdaTriggerHandler<TEvent, TRes>(input, context, scope))
                 {
                     handler.HandleBefore();
-
+                    setTraceUrl(handler.getTraceUrl());
```
Comment (tvaintrob) [inline src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs:47]: access local methods using `this` for consistency

**Review by tvaintrob — CHANGES_REQUESTED** [review]

File: src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs (hunk 2 above)
Comment (idonava) [reply src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs]: fixed

File: src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs (hunk 4 above)
Comment (idonava) [reply src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs]: I can't do it from a static method, fixed on LambdaTriggerHandler.

File: src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs (hunk 2 above)
Comment (idonava) [reply src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs]: fixed

File: src/Epsagon.Dotnet.Lambda/LambdaTriggerHandler.cs (hunk 3 above)
Comment (idonava) [reply src/Epsagon.Dotnet.Lambda/LambdaTriggerHandler.cs]: fixed for traceUrl, I can't do it for the local functions var.

File: src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs (hunk 1 above)
Comment (idonava) [reply src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs]: fixed

File: src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs [hunk 5]
```diff
@@ -116,6 +114,7 @@ public class EpsagonHandler
                 {
                     Utils.DebugLogIfEnabled("handling before execution");
                     handler.HandleBefore();
+                    SetTraceUrl(handler.GetTraceUrl());
```
Comment (tvaintrob) [inline src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs:117]: prefix local members/methods with `this`

File: src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs [hunk 6]
```diff
@@ -28,20 +31,18 @@ public class EpsagonHandler
...
                     trigger.Handle(context, scope);
                 }
-
                 // handle invocation event
                 using (var scope = GlobalTracer.Instance.BuildSpan((typeof(TEvent).Name)).StartActive(finishSpanOnDispose: true))
                 using (var handler = new LambdaTriggerHandler<TEvent, TRes>(input, context, scope))
                 {
                     handler.HandleBefore();
-
+                    SetTraceUrl(handler.GetTraceUrl());
```
Comment (tvaintrob) [inline src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs:45]: `this`

File: src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs [hunk 7]
```diff
@@ -194,6 +193,7 @@ public static async Task Handle<TEvent>(TEvent input, ILambdaContext context, Fu
                 {
                     Utils.DebugLogIfEnabled("handling before execution");
                     handler.HandleBefore();
+                    SetTraceUrl(handler.GetTraceUrl());
```
Comment (tvaintrob) [inline src/Epsagon.Dotnet.Lambda/EpsagonHandler.cs:196]: `this`

**Review by tvaintrob — CHANGES_REQUESTED** [review]

**Review by tvaintrob — APPROVED** [review]

**Pull request merged**